### DISCUSSION TITLE
v1.5.5

### DISCUSSION THREAD

**Discussion by ggerganov** [ggml-org/whisper.cpp#2064] [Announcements]
## Overview

Many small incremental updates + Token level timestamps with DTW by @denersc in https://github.com/ggerganov/whisper.cpp/pull/1485
Feedback is welcome!

**Full Changelog**: https://github.com/ggerganov/whisper.cpp/compare/v1.5.4...v1.5.5

## What's Changed
* server : fix server temperature + add temperature_inc by @ggerganov in https://github.com/ggerganov/whisper.cpp/pull/1729
* main : add cli option to disable system prints by @ggerganov in https://github.com/ggerganov/whisper.cpp/pull/1740
* server: add request path by @eschmidbauer in https://github.com/ggerganov/whisper.cpp/pull/1741
* Optional Piper TTS support for talk-llama example. by @RhinoDevel in https://github.com/ggerganov/whisper.cpp/pull/1749
* fix/1748 by @nank1ro in https://github.com/ggerganov/whisper.cpp/pull/1750
* Don't compute timestamps when not printing them. by @ghindle in https://github.com/ggerganov/whisper.cpp/pull/1755
* Add more parameters to server api by @ghindle in https://github.com/ggerganov/whisper.cpp/pull/1754
* Add SetInitialPrompt method to go bindings by @blib in https://github.com/ggerganov/whisper.cpp/pull/1753
* ggml : fix 32-bit ARM compat for IQ2_XS by @ggerganov in https://github.com/ggerganov/whisper.cpp/pull/1758
* refactor: get all scripts to be POSIX Compliant by @sonphantrung in https://github.com/ggerganov/whisper.cpp/pull/1725
* whisper : load the model into multiple buffers of max size 1GB by @ggerganov in https://github.com/ggerganov/whisper.cpp/pull/1763
* rebase against your -np changes (thx) and add better python file to be used on the command line or as library by @contractorwolf in https://github.com/ggerganov/whisper.cpp/pull/1744
* examples/talk-llama: Add optional commandline parameter to set the bot name. by @RhinoDevel in https://github.com/ggerganov/whisper.cpp/pull/1764
* server : fix building and simplify lib deps on Windows by @przemoc in https://github.com/ggerganov/whisper.cpp/pull/1772
* talk-llama: optional wake-up command and audio confirmation by @Rakksor in https://github.com/ggerganov/whisper.cpp/pull/1765
* examples/server: implement "verbose_json" format with token details by @rmmh in https://github.com/ggerganov/whisper.cpp/pull/1781
* whisper.android: Return output from benchmarks by @luciferous in https://github.com/ggerganov/whisper.cpp/pull/1785
* libwhisper.so should be position independent by @trixirt in https://github.com/ggerganov/whisper.cpp/pull/1792
* Docs: try to make model options / model install methods clearer by @mrienstra in https://github.com/ggerganov/whisper.cpp/pull/1806
* common : fix input buffer check by @ggerganov in https://github.com/ggerganov/whisper.cpp/pull/1812
* Update Makefile by @jwijffels in https://github.com/ggerganov/whisper.cpp/pull/1813
* Add fields to `verbose_json` response and show examples on the home page by @JacobLinCool in https://github.com/ggerganov/whisper.cpp/pull/1802
* common: fix wav buffer detection by @JacobLinCool in https://github.com/ggerganov/whisper.cpp/pull/1819
* Add macOS deployment target option to Makefile by @didzis in https://github.com/ggerganov/whisper.cpp/pull/1839
* Expose CUDA device setting in public API by @didzis in https://github.com/ggerganov/whisper.cpp/pull/1840
* whisper.android: How to build with CLBlast by @luciferous in https://github.com/ggerganov/whisper.cpp/pull/1809
* server: Allow CORS request with authorization headers by @valenting in https://github.com/ggerganov/whisper.cpp/pull/1850
* Embed Metal library source into compiled binary by @didzis in https://github.com/ggerganov/whisper.cpp/pull/1842
* added audio_ctx argument to main and server examples by @dscripka in https://github.com/ggerganov/whisper.cpp/pull/1857
* whisper : fix external encoder by @ggerganov in https://github.com/ggerganov/whisper.cpp/pull/1860
* swift : package no longer use ggml dependency by @ggerganov in https://github.com/ggerganov/whisper.cpp/pull/1861
* fix openvino setup docs by @jumpers775 in https://github.com/ggerganov/whisper.cpp/pull/1874
* clean up common code in examples by @felrock in https://github.com/ggerganov/whisper.cpp/pull/1871
* main : check if input files exist before proceeding by @Theldus in https://github.com/ggerganov/whisper.cpp/pull/1872
* Linking issue fix via Makefile when CUBLAS enabled in the WSL #1876 by @lbluep in https://github.com/ggerganov/whisper.cpp/pull/1878
* main : fix file existence check in main.cpp by @Theldus in https://github.com/ggerganov/whisper.cpp/pull/1889
* openvino : fix convert-whisper-to-openvino.py for v2023.0.0 (#1870) by @st-gr in https://github.com/ggerganov/whisper.cpp/pull/1890
* ggml : 32-bit arm compat by @ggerganov in https://github.com/ggerganov/whisper.cpp/pull/1891
* Add SYCL logic in whisper by @abhilash1910 in https://github.com/ggerganov/whisper.cpp/pull/1863
* talk and talk-llama: Pass text_to_speak as a file by @tamo in https://github.com/ggerganov/whisper.cpp/pull/1865
* Stream.wasm: Fix invalid memory access when no segments are returned by @Andrews54757 in https://github.com/ggerganov/whisper.cpp/pull/1902
* Update README to Recommend MacOS Sonoma for Core ML to avoid hallucination by @gavin1818 in https://github.com/ggerganov/whisper.cpp/pull/1917
* Add library versioning by @kenneth-ge in https://github.com/ggerganov/whisper.cpp/pull/1352
* Fix SF(segment fault) issue in Android JNI by @zhouwg in https://github.com/ggerganov/whisper.cpp/pull/1929
* Fix typo in source file whisper.cpp by @zhouwg in https://github.com/ggerganov/whisper.cpp/pull/1925
* bench:fix typo by @zhouwg in https://github.com/ggerganov/whisper.cpp/pull/1933
* Auto lowercase language parameter by @F1L1Pv2 in https://github.com/ggerganov/whisper.cpp/pull/1928
* ggml : try fix 32-bit arm compat by @ggerganov in https://github.com/ggerganov/whisper.cpp/pull/1938
* whisper : make beam candidate sort more stable by @josharian in https://github.com/ggerganov/whisper.cpp/pull/1943
* bindings/go : add linker flags to make metal work by @josharian in https://github.com/ggerganov/whisper.cpp/pull/1944
* whisper : improve beam search candidate diversity by @josharian in https://github.com/ggerganov/whisper.cpp/pull/1947
* whisper : document whisper_batch.n_seq_id by @josharian in https://github.com/ggerganov/whisper.cpp/pull/1942
* Rename --audio-context to --audio-ctx, as per help text by @joliss in https://github.com/ggerganov/whisper.cpp/pull/1953
* [DRAFT] Token level timestamps with DTW (#375)  by @denersc in https://github.com/ggerganov/whisper.cpp/pull/1485
* Fedora dependencies needed (SDL2) by @Man2Dev in https://github.com/ggerganov/whisper.cpp/pull/1970
* libcuda.so.1 in PATH in Docker Container by @tiagofassoni in https://github.com/ggerganov/whisper.cpp/pull/1966
* ruby : fix build by @ggerganov in https://github.com/ggerganov/whisper.cpp/pull/1980
* Improve support for distil-large-v3 by @sanchit-gandhi in https://github.com/ggerganov/whisper.cpp/pull/1982
* whisper : improve handling of prompts by @ggerganov in https://github.com/ggerganov/whisper.cpp/pull/1981
* sync : ggml by @ggerganov in https://github.com/ggerganov/whisper.cpp/pull/2001
* Implemented command-style grammar in the main example. by @ulatekh in https://github.com/ggerganov/whisper.cpp/pull/1998
* Use pkg-config for OpenBLAS by @przemoc in https://github.com/ggerganov/whisper.cpp/pull/1778
* ci : add building in MSYS2 environments (Windows) by @przemoc in https://github.com/ggerganov/whisper.cpp/pull/1994
* Support CUDA versions < 11.1 by @primenko-v in https://github.com/ggerganov/whisper.cpp/pull/2020
* Create solution folders in the CMake build by @ulatekh in https://github.com/ggerganov/whisper.cpp/pull/2004
* Allow a regular expression to describe tokens to suppress by @ulatekh in https://github.com/ggerganov/whisper.cpp/pull/1997
* "main" example now allows a response-file as the sole parameter by @ulatekh in https://github.com/ggerganov/whisper.cpp/pull/2019
* Support for CPU BLAS build via Intel MKL by @slashlib in https://github.com/ggerganov/whisper.cpp/pull/2024
* Set stdin to binary mode on Windows. Fixes #2023 by @rotemdan in https://github.com/ggerganov/whisper.cpp/pull/2025
* Fix file-handle leak in read_wav() by @ulatekh in https://github.com/ggerganov/whisper.cpp/pull/2026
* Fix DTW memory access by @bradmurray-dt in https://github.com/ggerganov/whisper.cpp/pull/2012
* whisper: update grammar-parser.cpp by @eltociear in https://github.com/ggerganov/whisper.cpp/pull/2058
* fix missing reference to "model" variable in actual shell command run in whisper.nvim by @sixcircuit in https://github.com/ggerganov/whisper.cpp/pull/2049
* build : detect AVX512 in Makefile, add AVX512 option in CMake by @didzis in https://github.com/ggerganov/whisper.cpp/pull/2043
* feature/no timestamps node by @pprobst in https://github.com/ggerganov/whisper.cpp/pull/2048
* Update embedded Metal library generation process to include dependency by @didzis in https://github.com/ggerganov/whisper.cpp/pull/2045
* server.cpp: add dtw by @eschmidbauer in https://github.com/ggerganov/whisper.cpp/pull/2044

## New Contributors
* @eschmidbauer made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/1741
* @RhinoDevel made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/1749
* @nank1ro made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/1750
* @ghindle made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/1755
* @blib made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/1753
* @sonphantrung made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/1725
* @contractorwolf made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/1744
* @Rakksor made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/1765
* @rmmh made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/1781
* @luciferous made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/1785
* @trixirt made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/1792
* @mrienstra made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/1806
* @JacobLinCool made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/1802
* @valenting made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/1850
* @dscripka made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/1857
* @jumpers775 made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/1874
* @Theldus made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/1872
* @lbluep made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/1878
* @st-gr made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/1890
* @abhilash1910 made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/1863
* @Andrews54757 made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/1902
* @gavin1818 made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/1917
* @kenneth-ge made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/1352
* @zhouwg made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/1929
* @F1L1Pv2 made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/1928
* @josharian made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/1943
* @joliss made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/1953
* @Man2Dev made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/1970
* @tiagofassoni made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/1966
* @sanchit-gandhi made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/1982
* @ulatekh made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/1998
* @primenko-v made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/2020
* @slashlib made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/2024
* @rotemdan made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/2025
* @bradmurray-dt made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/2012
* @sixcircuit made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/2049
* @pprobst made their first contribution in https://github.com/ggerganov/whisper.cpp/pull/2048

**Full Changelog**: https://github.com/ggerganov/whisper.cpp/compare/v1.5.4...v1.5.5

<hr /><em>This discussion was created from the release <a href='https://github.com/ggerganov/whisper.cpp/releases/tag/v1.5.5'>v1.5.5</a>.</em>

**Comment by timwu-ipevo** [comment]
Will executable files built for various accelerations also be released like past releases?

**Reply by ggerganov** [reply]
There was a last-minute regression in `1.5.5` (https://github.com/ggerganov/whisper.cpp/issues/2066) causing extra CPU usage in the `main` example. After fixing it https://github.com/ggerganov/whisper.cpp/pull/2070 I'm thinking of making a new minor release and will add the binaries there. Will give this a day or two before `1.5.6` to see if there are any other problems present

**Reply by SunnyOd** [reply]
Hi @ggerganov 

I can see you've been adding binaries to the releases. Looks like they're only for windows though? Any chance of adding Linux binaries too? CUDA toolkit is a nightmare to install on so many distro's. I'm running Fedora so I am forced to use the CPU binaries as I can compile those okay

Btw thanks for this project it saves me hours every day, appreciate it!

**Reply by ggerganov** [reply]
AFAIAW creating binaries for anything else than Windows is quite tricky, so I don't think we will add this soon. Though, if there are contributions to make this work in a reasonable way, we can do it

**Reply by ulatekh** [reply]
@SunnyOd : I build whisper.cpp with CUDA under Fedora just fine. After installing the relevant [CUDA repo](https://rpmfusion.org/Howto/CUDA), you'll need to build and install [GCC 12](https://reddit.com/comments/12xfowq/-/kyh8gdv/).

**Comment by uckthis** [comment]
We use it in Subtitle Edit - it's the most efficient of all the engines available. Waiting for new EXE, though.

Thank you.

**Reply by ggerganov** [reply]
I'll focus the next day or two on `whisper.cpp` issues and try to push a `v1.5.6` release with Win binaries

**Reply by ggerganov** [reply]
Plan is to release `v1.6.0` with Flash Attention support today/tomorrow. Feedback on the new FA functionality is appreciated: https://github.com/ggerganov/whisper.cpp/pull/2152
